app/src/main/java/com/kms/cura/view/fragment/RegDoc_AccountFragment.java

```diff
@@ -0,0 +1,181 @@
+            Bundle bundle = createBundle();
+            Fragment professional = new RegDoc_ProfessionalFragment();
+            professional.setArguments(bundle);
+            getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, professional).commit();
+        } else if (v.getId() == R.id.btnRegister) {
+            if (checkInput()) {
+                String phone = edtPhone.getText().toString();
+                String email = edtEmail.getText().toString();
+                String pwd = edtPwd.getText().toString();
+                Bundle bundle = getArguments();
+                // check in database
+
+            }
+        }
+    }
+
+    private boolean checkInput() {
+        return checkEmail() && checkPwd() && checkRePwd() && checkPhone();
+    }
+
+    private boolean checkPhone() {
+        return !edtPhone.getText().toString().isEmpty();
+    }
+
+    private boolean checkEmail() {
+        String email = edtEmail.getText().toString();
+        return InputUtils.isEmailValid(email);
+    }
+
+    private boolean checkPwd() {
+        String pwd = edtPwd.getText().toString();
+        return InputUtils.isPasswordValid(pwd);
+    }
+
+    private boolean checkRePwd() {
+        String pwd = edtPwd.getText().toString();
+        String rePwd = edtRePwd.getText().toString();
+        return pwd.equals(rePwd);
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        flag = true;
+
+        setErrorMessagePhone();
+
+        setErrorMessage(edtEmail, edittedEmail, getString(R.string.email_error));
+
+        setErrorMessage(edtPwd, edittedPwd, getString(R.string.pwd_error));
+
+        setErrorMessageRePwd();
+
+        btnRegister.setEnabled(flag);
+    }
+
+    @Override
+    public void afterTextChanged(Editable s) {
+
+    }
+
+    private Bundle createBundle() {
+        Bundle bundle = getArguments();
+        bundle.putBoolean(RegisterDoctorActivity.PROFESSIONAL, true);
+        bundle.putString(RegisterDoctorActivity.PHONE, edtPhone.getText().toString());
+        bundle.putString(RegisterDoctorActivity.EMAIL, edtEmail.getText().toString());
+        return bundle;
+    }
+
+    private void reformData() {
+        Bundle bundle = getArguments();
+        edtPhone.setText(bundle.getString(RegisterDoctorActivity.PHONE, ""));
+        edtEmail.setText(bundle.getString(RegisterDoctorActivity.EMAIL, ""));
+        edittedPhone = !edtPhone.getText().toString().equals("");
+        edittedEmail = !edtEmail.getText().toString().equals("");
+    }
+
+    private void setErrorMessage(EditText edt, boolean editted, String errorMes) {
+        if (edt.getText().toString().equals("")) {
+            if (editted)
+                edt.setError(getString(R.string.required_field));
+            editted = false;
+            flag = false;
```

Comment:
Don't make flag into a class field. Instead change this method into return a boolean to indicate the status.